api/graph_test.go

```diff
@@ -0,0 +1,64 @@
+package api
```

Comment:
You should test that this is as well a builder pattern. All your tests have a length of one.

Reply:
I don't understand this point.

Reply:
In case of the vertex you did a test, that used the builder pattern. I think you should add one for the graph as well.